src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -1535,18 +1535,20 @@ class OpenShiftDSL implements Serializable {
             HashMap<String, String> projList = new HashMap<String, String>();
             List<String> names = names();
             HashMap<String, String> projects = projects();
+            // leverage startsWith in comparisons to handle the new kinds like
+            // deploymentconfig.apps.openshift.io
```

Comment:
do we only see this new output w/ the new oc binary?  or does the old oc binary return these values when talking to a new api server?

Reply:
Well the `oc` in the image is at `oc v3.10.0-alpha.0+0792236-777`

Also, when I exec into the pod, and run loglevel=10, what I see is:

```
{"kind":"DeploymentConfig","apiVersion":"apps.openshift.io/v1"
```

instead of the traditional `v1` ... similar with builds as well ... the plugin's dsl string parsing must be picking the `apps.openshift.io` from there somehow, vs. my original interpretation (I did not debug and the raw level).

I can go back and confirm that .... it may make sense to redo the current parsing to account for it there if that is where it is coming from.

Reply:
OK, I figured out where the new names were coming from ... it is the `-o name` option that this plugin employs to confirm the existence of selectors.

Here is some sample output from within my jenkins pod, to confirm the same occurs with calling `oc`:

```
sh-4.2$ oc get dc jenkins
NAME      REVISION   DESIRED   CURRENT   TRIGGERED BY
jenkins   1          1         1         config
sh-4.2$ oc get dc jenkins -o name
deploymentconfig.apps.openshift.io/jenkins
sh-4.2$ oc get dc jenkins -o yaml
apiVersion: apps.openshift.io/v1
kind: DeploymentConfig
metadata:
  annotations:
    openshift.io/generated-by: OpenShiftNewApp
    template.alpha.openshift.io/wait-for-ready: "true"
  creationTimestamp: 2018-04-20T21:03:42Z
  generation: 1
  labels:
    app: jenkins-ephemeral
    template: jenkins-ephemeral-template
  name: jenkins
  namespace: myproject
  resourceVersion: "4006"
  selfLink: /apis/apps.openshift.io/v1/namespaces/myproject/deploymentconfigs/jenkins
  uid: 4eb9ad30-44de-11e8-8c85-54e1addf5a37
spec:
....
```

Note that with `-o yaml` we don't see that string ... that is what was confusing me.

To answer you original question @bparees, I downloaded the 3.9 oc, ran it against the same cluster, and `-o name` in fact did return the old `deploymentconfigs` form:

```
gmontero ~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-linux-64bit $ ./oc version
oc v3.9.0+191fece
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://127.0.0.1:8443
openshift v3.10.0-alpha.0+45d2bb2-816
kubernetes v1.10.0+b81c8f8
gmontero ~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-linux-64bit $ ./oc get dc jenkins 
NAME      REVISION   DESIRED   CURRENT   TRIGGERED BY
jenkins   1          1         1         config
gmontero ~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-linux-64bit $ ./oc get dc jenkins -o name
deploymentconfigs/jenkins
gmontero ~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-linux-64bit $ 
```

Reply:
So it is a recent change in the new 3.10 client

Reply:
good, that's at least manageable.